### PR TITLE
WD-10181 - feature: add entitlements when creating a role

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add entitlements panel.
- Submit the request to add entitlements once the role has been created.
- Update openapi.yaml to add the entitlements patch request.
- 

## QA

- Connect to the real API.
- Open the 'Create role' panel.
- Fill out the role name.
- Click "Add entitlements".
- Fill out the form. Some values need to be value so you can use something like: 
  - Resource type: "client"
  - Resource: "aws"
  - Entitlement: "can_view"
  - Click the "Create role" link at the top to go back to the first screen.
  - Click the "Create role" button and the role should get added.
  - If you want to see that the entitlements were created you can check the output of `fga tuple read --simple-output --api-url http://127.0.0.1:8080 --store-id $STORE_ID`

## Details

- https://warthogs.atlassian.net/browse/WD-10181

## Screenshots

Screenshot 2024-04-11 at 3 52 36 PM

### CODE REVIEW

File: openapi.yaml
Comment (huwshimi) [inline openapi.yaml:1]: The changes in this file have been copied from https://github.com/canonical/rebac-admin/pull/82.

File: src/components/EntitlementsPanelForm/EntitlementsPanelForm.tsx
Comment (huwshimi) [inline src/components/EntitlementsPanelForm/EntitlementsPanelForm.tsx:1]: This form will get reused by the other panels e.g. when adding groups.

File: src/components/PanelForm/PanelForm.tsx [hunk 1]
```diff
@@ -18,52 +23,80 @@ const PanelForm = <F extends FormikValues>({
...
-            <Col size={12}>
-              <Notification severity={NotificationSeverity.NEGATIVE}>
-                {error}
-              </Notification>
-            </Col>
-          </Row>
-        ) : null}
-        <Formik<F> {...props}>
-          <Form>
+        {view
```
Comment (huwshimi) [inline src/components/PanelForm/PanelForm.tsx:45]: These sub-forms are the secondary panel views e.g. adding entitlements to a role or group.

File: src/components/PanelForm/PanelForm.tsx [hunk 2]
```diff
@@ -18,52 +23,80 @@ const PanelForm = <F extends FormikValues>({
...
-        </Formik>
+          ) : null}
+          <Formik<F> {...props}>
+            <Form>
+              <Row>
+                <Col size={12}>{children}</Col>
+              </Row>
+              <Row>
+                <Col size={12}>
+                  {subForms.map(({ count, entity, icon }) => (
```
Comment (huwshimi) [inline src/components/PanelForm/PanelForm.tsx:72]: This generates the links to navigate to the sub forms.

File: src/components/PanelForm/PanelFormLink/PanelFormLink.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { Button, Icon } from "@canonical/react-components";
+
+import { type Props } from "./types";
+import { generateTitle } from "./utils";
+
+import "./_panel-form-link.scss";
+
+const PanelFormLink = ({ count, entity, icon, isEditing, onClick }: Props) => {
```
Comment (huwshimi) [inline src/components/PanelForm/PanelFormLink/PanelFormLink.tsx:8]: These are the links to the sub forms.

File: src/components/PanelForm/PanelFormNavigation/PanelFormNavigation.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { Button, Icon } from "@canonical/react-components";
+
+import { generateTitle } from "../PanelFormLink/utils";
+
+import { PanelFormNavigationTitleId } from "./consts";
+import { type Props } from "./types";
+
+import "./_panel-form-navigation.scss";
+
+const PanelFormNavigation = ({
```
Comment (huwshimi) [inline src/components/PanelForm/PanelFormNavigation/PanelFormNavigation.tsx:10]: This navigation is used to navigate back to the initial panel view.

**Review by Lukewh — APPROVED** [review]
:+1: LGTM

**Comment by Lukewh** [comment]
Code looks good. I haven't QA'd though!

**Review by vladimir-cucu — APPROVED** [review]
I have QA'd against the real API. Followed the QA steps and everything worked as expected! :+1: The following was created:
```
{
    "object":"client:aws",
    "relation":"can_view",
    "user":"role:new-role#assignee"
  }
  ```

**Pull request merged**